runtime/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/stream/MqttKafkaSubscribeFactory.java

```diff
@@ -193,7 +199,7 @@ private final class MqttSubscribeProxy
         private final long routedId;
         private final long initialId;
         private final long replyId;
-        private final KafkaMessagesProxy messages;
+        private final Object2ObjectHashMap<MqttKafkaRouteConfig, KafkaMessagesProxy> messages;
```

Comment:
Do we need to key by route config here?
I think each route knows its own index, so using the route index and/or a bitmask of matching route indexes might be lighter.

Reply:
Good idea. I guess route.order is what I'm looking for